vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -282,17 +289,98 @@ def parse_container_definition(container_def, pod_id)
 
     def parse_container(container, pod_id)
       {
-        :type          => 'ContainerKubernetes',
-        :ems_ref       => "#{pod_id}_#{container.name}_#{container.image}",
-        :name          => container.name,
-        :image         => container.image,
-        :restart_count => container.restartCount,
-        :backing_ref   => container.containerID,
-        :image_ref     => container.imageID
+        :type            => 'ContainerKubernetes',
+        :ems_ref         => "#{pod_id}_#{container.name}_#{container.image}",
+        :name            => container.name,
+        :image           => container.image,
+        :restart_count   => container.restartCount,
+        :backing_ref     => container.containerID,
+        :image_ref       => container.imageID,
+        :container_image => parse_container_image(container.image, container.imageID)
       }
+
       # TODO, state
     end
 
+    def parse_container_image(image, imageID)
+
+      # Regex is unlikely to be the best way to do this. This might take a while with 10,000 containers...
+      # 'name' or 'user/name'
```

Comment:
Drop the regexes. Also, you're not handling images as:

    gcr.io/google_containers/etcd:2.0.9

And remember that the repository port can be optional.